### PR TITLE
Add article about Lua as ESP-IDF Component

### PR DESCRIPTION
**Pull request by georgik**
The article contains explanation how to get Lua working with an ESP-IDF application including examples, how to execute Lua scripts stored in files.

### CODE REVIEW

File: content/blog/using-lua-as-esp-idf-component-with-esp32/index.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    lua_close(L);
+    ESP_LOGI(TAG, "Finished running Lua script: %s", description);
+}
+```
+
+Creates a new Lua state and opens standard libraries.
+Sets the package.path to include the filesystem path where Lua scripts are stored.
+Executes the Lua script from the specified file.
```
Comment (f-hollow) [inline content/blog/using-lua-as-esp-idf-component-with-esp32/index.md:92]: These sentences read like code comments but not standalone sentences. How about adding them as comments?

File: content/blog/using-lua-as-esp-idf-component-with-esp32/index.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    } else {
+        ESP_LOGE(TAG, "Failed to get SPIFFS partition information");
+    }
+}
+```
+
+
+### Getting Started
+
+Clone the Repository
```
Comment (f-hollow) [inline content/blog/using-lua-as-esp-idf-component-with-esp32/index.md:129]: Looks like these are steps to complete to get started.

Why not numbering them or formatting them as separate steps in some other ways?

File: content/blog/using-lua-as-esp-idf-component-with-esp32/index.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+```shell
+The answer is: 42
+Fibonacci of 10 is: 55
+QR Code for: https://developer.espressif.com/tags/lua
+```
+
+
+## Integrating Lua as an ESP-IDF Component
+
+To incorporate Lua into your ESP32 project seamlessly, you can use the ESP-IDF Component Registry, which simplifies dependency management and ensures you're using a stable version of Lua.
```
Comment (f-hollow) [inline content/blog/using-lua-as-esp-idf-component-with-esp32/index.md:157]: See the comment about _seamlessly_ above.

File: content/blog/using-lua-as-esp-idf-component-with-esp32/index.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  joltwallet/littlefs: "==1.14.8"
+
+idf:
+  version: ">=5.0.0"
+```
+This configuration tells the ESP-IDF build system to include version 5.4.7 of the Lua component maintained by georgik, as well as version 1.14.8 of the LittleFS component for filesystem support. The idf section ensures compatibility with ESP-IDF version 5.0.0 or newer.
+
+### Utilizing the ESP-IDF Component Registry
+
+By specifying Lua in your idf_component.yml, the ESP-IDF build system automatically downloads and integrates the Lua component during the build process. There's no need to manually clone or manage the Lua source code.
```
Comment (f-hollow) [inline content/blog/using-lua-as-esp-idf-component-with-esp32/index.md:176]: Should we consider using monospace for file paths and names?

```suggestion
By specifying Lua in your `idf_component.yml`, the ESP-IDF build system automatically downloads and integrates the Lua component during the build process. There's no need to manually clone or manage the Lua source code.
```

If yes, suggest updating the other file paths in names in this article.

**Review by f-hollow — APPROVED** [review]
The article is crisp and straight to the point!

I have a few minor suggestions here and there to possibly further sharpen its appearance.

**Review by f-hollow — APPROVED** [review]

**Review by pedrominatel — APPROVED** [review]

**Pull request merged**